atoma-state/src/state_manager.rs

```diff
@@ -1509,23 +1518,24 @@ impl AtomaState {
         stack_small_id: i64,
         estimated_total_compute_units: i64,
         total_compute_units: i64,
-    ) -> Result<()> {
+    ) -> Result<(f64, i64)> {
         let result = sqlx::query(
             "UPDATE stacks
             SET already_computed_units = already_computed_units - ($1 - $2)
-            WHERE stack_small_id = $3",
+            WHERE stack_small_id = $3
+            RETURNING already_computed_units, 
+                CAST(already_computed_units AS FLOAT) / CAST(num_compute_units AS FLOAT)",
         )
         .bind(estimated_total_compute_units)
         .bind(total_compute_units)
         .bind(stack_small_id)
-        .execute(&self.db)
+        .fetch_optional(&self.db)
         .await?;
 
-        if result.rows_affected() == 0 {
-            return Err(AtomaStateManagerError::StackNotFound);
-        }
-
-        Ok(())
+        result.map_or_else(
+            || Err(AtomaStateManagerError::StackNotFound),
+            |ratio| Ok((ratio.get(1), ratio.get(0))),
```

Comment:
I mean this order changing
it's easier to read the function when the order doesn't change